### DISCUSSION TITLE
lesson11 MoodNftIntegrationTest's testFlipTokenToSad fail

### DISCUSSION THREAD

**Discussion by RaferGuo** [Cyfrin/foundry-full-course-cu#1527] [Q&A] [answered]
here‘s my repo：https://github.com/RaferGuo/foundry-nft-test
actually，i didn't know how patrick decode this
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/5eccaffb-344e-4b3c-9551-4721b3c9f240) to 
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/e47ed2ec-0445-4f3a-b4ac-05fe127789a2)
and then this (data:application format) in video 9：28：30 nearby.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/f52138dc-d612-468b-ac25-96f40cd1e4ca)
so i copy the’ code SAD_SVG_URI from cyfrin's repo.
how can i fix it, thx
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/748d38a5-4f87-484f-8140-b1f46aca049d)

**Comment by EngrPips** [comment, marked as answer]
Hello @RaferGuo , The easiest way to fix this would be to copy the `moodNft::tokenURI` that is returned from your `MoodNFT` contract [That is what is printed towards the end of the of the screenshot you sent] and set it as the `expectedtokenURI` instead of copying Patrick `expectedTokenURI`, Like that your test should pass. base64 encoding is very sensitive to any change so any slightest different from what Patrick did to get is `expectedURI` would make your own actual URI totally different from Patrick `expectedURI`
